src/main/java/com/gooddata/auditevent/AuditEvent.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2004-2017, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.auditevent;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.gooddata.util.GoodDataToStringBuilder;
+import com.gooddata.util.ISODateTimeDeserializer;
+import com.gooddata.util.ISODateTimeSerializer;
+import org.joda.time.DateTime;
+import org.springframework.web.util.UriTemplate;
+
+import java.util.Map;
+
+/**
+ * Audit event
+ */
+@JsonTypeInfo(include = JsonTypeInfo.As.WRAPPER_OBJECT, use = JsonTypeInfo.Id.NAME)
+@JsonTypeName(AuditEvent.ROOT_NODE)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class AuditEvent {
+
+    public static final String GDC_URI = "/gdc";
+    public static final String USER_URI = GDC_URI + "/account/profile/{userId}/auditEvents";
+    public static final String ADMIN_URI = GDC_URI + "/domains/{domainId}/auditEvents";
+
+    public static final UriTemplate ADMIN_URI_TEMPLATE = new UriTemplate(ADMIN_URI);
+    public static final UriTemplate USER_URI_TEMPLATE = new UriTemplate(USER_URI);
+
+    static final String ROOT_NODE = "event";
+
+    private final String id;
+
+    private final String userLogin;
+
+    private final DateTime occurred; //time event happened at component
+
+    private final DateTime recorded; //time of insertion to mongo
```

Comment:
What about to replace "mongo" with "audit event storage"? And it would be better to move these comments to javadocs of accessor methods.

Reply:
added to javadoc